### ISSUE TITLE
Historian does not find objects if there is not a save first

### ISSUE THREAD

**Issue by edeguine** [muhrin/mincepy#20]
```
import mincepy

from service.config.config import Config
config = Config()
historian = mincepy.connect(config.mongo_uri, use_globally=True)

class User(mincepy.SimpleSavable):
    TYPE_ID = 'User'
    name = mincepy.field()
    email = mincepy.field()
    api_key = mincepy.field()


my_user = User(name="Etienne", email='hello@email.com', api_key='123')
#historian.save(my_user)
my_user = User(name="Etienne", email='hello@email.com', api_key='124')
#historian.save(my_user)

same_user = historian.find(User, state=mincepy.q.and_({'api_key': '123'}, {'email': 'hello@email.com'}))
for res in same_user:
    print(res.api_key)
```
    
If i uncomment the save lines everything runs fine. But if i comment them and rerun the script, without touching the DB (so the data is still there), the historian fails to find anything with this exception:

```
Traceback (most recent call last):
  File "/Users/etienne/sdk/src/domain/data/ORM_manager.py", line 20, in <module>
    for res in same_user:
  File "/Users/etienne/fluffyenv/lib/python3.9/site-packages/mincepy/frontend.py", line 44, in __iter__
    yield self._entry_factory(entry)
  File "/Users/etienne/fluffyenv/lib/python3.9/site-packages/mincepy/frontend.py", line 261, in _create_object
    return self._obj_loader(self._record_factory(archive_record))
  File "/Users/etienne/fluffyenv/lib/python3.9/site-packages/mincepy/historians.py", line 911, in _load_object_from_record
    return depositor.load_from_record(record)
  File "/Users/etienne/fluffyenv/lib/python3.9/site-packages/mincepy/depositors.py", line 215, in load_from_record
    loaded = self.decode(record.state,
  File "/Users/etienne/fluffyenv/lib/python3.9/site-packages/mincepy/depositors.py", line 132, in decode
    helper = self.get_historian().get_helper(entry.type_id)
  File "/Users/etienne/fluffyenv/lib/python3.9/site-packages/mincepy/historians.py", line 538, in get_helper
    return self._type_registry.get_helper(type_id_or_type)
  File "/Users/etienne/fluffyenv/lib/python3.9/site-packages/mincepy/type_registry.py", line 60, in get_helper
    return self.get_helper_from_type_id(type_id_or_type)
  File "/Users/etienne/fluffyenv/lib/python3.9/site-packages/mincepy/type_registry.py", line 66, in get_helper_from_type_id
    raise TypeError(f"Type id '{type_id}' not known") from None
TypeError: Type id 'User' not known
```

I did not see anything in the documentation about this, am I supposed to call historian.load_something() first?

**Comment by muhrin** [comment]
Hi @edeguine , thanks for posting - sorry I hadn't seen this up till now.

The TLDR is that, if you have a python file where you haven't yet saved a type then you need to register it with the Historian using:
```
historian.register_type(User)
```
before calling `find()` (or any other method that relies on the type being understood).  
Now, in this case I think it would be more useful if the Historian automatically called this for you when you call `find()` so I will shortly put out a patch release with this (you can track this in #21 ).

The longer answer is that the Historian doesn't automatically search python files for compatible types (in this case `User`) because different code bases may use the same `TYPE_ID` (for example `User` would certainly be a common choice) and if you were using a third-party library that also used the same ID the historian wouldn't be able to choose which one to use.  So, to be safe, by default users need to manually register.  This is one of the reasons that in my examples I just use UUIDs as type ids.  There is also a plugin system that allows mincePy to call a method of your code to discover which types you want to use.  This has not been documented and I have opened an issue here #22 to track progress.

As a final note, maybe you've already seen this, but you can make your `find` call more human readable by using:

```
same_user = historian.find(User.api_key == '123', User.email == 'hello@email.com')
```

**Comment by muhrin** [comment]
Ok, this should be fixed in [v0.15.28](https://github.com/muhrin/mincepy/releases/tag/v0.15.28.)

Would you mind checking if your problem is resolved?

**Comment by muhrin** [comment]
I'm going to close this issue as I'm guessing your problem is now solved @edeguine , but if not feel free to comment and I will re-open.
